### DISCUSSION TITLE
Support for Azure Open AI?

### DISCUSSION THREAD

**Discussion by ghost** [langchain-ai/open-canvas#180] [Q&A] [answered]
Hello, anyone have tips for transitioning all models to Azure Open AI and Azure AI studio based models?

**Comment by bracesproul** [comment, marked as answer]
Hey @nicoyoung-AS, we've added support for using Azure OpenAI!

To use Azure OpenAI instead of the OpenAI API, set these environment variables (note the prefixing underscore for Azure API keys is required):
```env
# Required for the Azure models to appear in the model dropdown
NEXT_PUBLIC_AZURE_ENABLED="true"

# Env variables for calling the Azure OpenAI API
_AZURE_OPENAI_API_KEY=your-azure-openai-api-key
_AZURE_OPENAI_API_INSTANCE_NAME=your-instance-name
_AZURE_OPENAI_API_DEPLOYMENT_NAME=your-deployment-name
_AZURE_OPENAI_API_VERSION=2024-08-01-preview
```
